### ISSUE TITLE
Dashboard Settings: Site Kit Specific UI

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#3809]
On the Dashboard Settings page we are allowing admins to update their Google Analytics ID. 
Per product specs: [`If Site Kit is installed, the user is informed that they don't have to enter the tracking ID here manually.](https://docs.google.com/document/d/1Xs4uOZ3KCNlTMdjDT6q7cvTUpA1xk-kzrnVLQALltQk/edit?pli=1#)`. 

Currently this spec has no UX.
How do we want to display this information in the page?  
How will information about if Site Kit is installed be gathered to the API and sent to the front end? 

Once this information is known, development can start.

**Comment by swissspidy** [comment]
**Note:** There can be two different states here:

1. Site Kit is active, but the Analytics module is not. For example when the user is only using the AdSense module.
1. Both Site Kit and the Analytics module are fully active and working.

Both is fairly trivial to detect.

Should we detect both cases and offer different information depending on the state? For example:

1. "Activate the Analytics module in Site Kit to automatically enable analytics for your Web Stories"
1. "Since you are using Site Kit, the analytics configuration will be handled entirely by Site Kit"

**Comment by o-fernandez** [comment]
> **Note:** There can be two different states here:
> 
> 1. Site Kit is active, but the Analytics module is not. For example when the user is only using the AdSense module.
> 2. Both Site Kit and the Analytics module are fully active and working.
> 
> Both is fairly trivial to detect.
> 
> Should we detect both cases and offer different information depending on the state? For example:
> 
> 1. "Activate the Analytics module in Site Kit to automatically enable analytics for your Web Stories"
> 2. "Since you are using Site Kit, the analytics configuration will be handled entirely by Site Kit"

+1 we should do that. 

@BrittanyIRL @carlos-kelly is there a basic version of this we could do in the dashboard without requiring a designer to create designs? I'd love for us to do this without being blocked on UX resources. We can always refine in the future. 

I'd do as Pascal mentioned, but add a 3rd scenario: 
3. Site Kit is not installed, help text under "Google Analytics Tracking ID" can say: "Use [Site Kit](https://wordpress.org/plugins/google-site-kit/) to easily configure Google Analytics."

**Comment by carloskelly13** [comment]
@o-fernandez Definitely! We can do a little design work that follows the flow. If Sam dislikes it we can iterate on it!

**Comment by BrittanyIRL** [comment]
@o-fernandez 
Just as a first draft on where the site kit updates would live - can I get input on the below? 
This would just update the text below the "Google Analytics Tracking Id" heading and leave the input and helper text below since it's still relevant to the story analytic set up regardless of site kit being active (is my understanding). I want to make sure I'm understanding the states of the page and what you have in mind before going further. 

Site Kit is not installed and obviously not active:
- analytics input is enabled, can be updated from settings view.
<img width="1091" alt="Screen Shot 2020-10-27 at 4 36 43 PM" src="https://user-images.githubusercontent.com/10720454/97373232-bd86d280-1872-11eb-89fd-bd006a8f4714.png">

Site Kit is installed and inactive:
- Since site kit is installed the analytics input is disabled and user is encouraged to set up analytics from site kit
<img width="1089" alt="Screen Shot 2020-10-27 at 4 36 55 PM" src="https://user-images.githubusercontent.com/10720454/97373235-c081c300-1872-11eb-9dff-c16dfbd863f1.png">


Site Kit is installed and active:
- Since site kit is installed the analytics input is disabled but analytics id that was set in site kit is visible in the disabled input.
<img width="1090" alt="Screen Shot 2020-10-27 at 4 37 05 PM" src="https://user-images.githubusercontent.com/10720454/97373233-beb7ff80-1872-11eb-93cd-ea7c7bc937d5.png">

**Comment by swissspidy** [comment]
Hmm personally I would have put it below the helper text.

> **Did you know?** Use [Site Kit by Google]() to automatically enable analytics for your Web Stories. [Install plugin]().

> **Did you know?** Use [Site Kit by Google]() to automatically enable analytics for your Web Stories. [Activate Analytics module]().

> **Did you know?** Since you are using [Site Kit by Google](), the analytics configuration will be handled entirely by Site Kit. So there's no need for you to do anything!

**Comment by BrittanyIRL** [comment]
@swissspidy Thanks! 

I'll try that out too today. I was more looking to make sure i wasn't missing some other state within the different statuses of site kit a user could have. Seems like I'm on the right trail though, I'll work on making this prettier today. 💅

**Comment by BrittanyIRL** [comment]
@o-fernandez iterated on the above with @swissspidy 's suggestions. Thoughts? 

Option 1: site kit info is below heading, separated from helper text: 
<img width="1084" alt="Screen Shot 2020-10-28 at 9 45 16 AM" src="https://user-images.githubusercontent.com/10720454/97468858-f9697880-1902-11eb-9b29-c9ecc1703986.png">


Option 2: Site kit info is below helper text. 
<img width="1079" alt="Screen Shot 2020-10-28 at 9 49 59 AM" src="https://user-images.githubusercontent.com/10720454/97468864-fa9aa580-1902-11eb-8a5f-58e941da6723.png">

**Comment by o-fernandez** [comment]
I actually like Option 1 more, now that I look at them, since there's also a pattern for other text under the headings below. I'd change the text to be a bit shorter and have only one link: 

- If not installed: **[Install Site Kit by Google](https://wordpress.org/plugins/google-site-kit/)** to easily enable Google Analytics for Web Stories.
- In installed and not active: Use Site Kit by Google to easily **activate Google Analytics** for Web Stories.
- if installed and active: Site Kit by Google has already enabled Google Analytics for your Web Stories, no need to do anything else.

@swissspidy what do you think about that? Also, which links do you think are most appropriate? I was thinking we can link either to the [Site Kit site with instructions to install](https://sitekit.withgoogle.com/documentation/install/)  or directly to the [WP plugin site
](https://wordpress.org/plugins/google-site-kit/)

**Comment by swissspidy** [comment]
SGTM 👍 

Regarding the links:

**If Site Kit is not installed:**

a) If the user has permissions to install plugins, provide direct link to install from within WordPress admin.\*
b) The WordPress plugin page. (so they can first read what Site Kit is instead of being presented with installation instructions)

**If Site Kit is installed but not active:**

Not sure we want to treat this different from the above scenario.

**If Site Kit is active, but GA module is not active**

a) If the user has permissions to configure Site Kit, provide direct link to Sit Kit settings screen.
b) If user lacks these permissions (quite unlikely but possible), either provide link to Site Kit dashboard, or provide no link and write something like "Reach out to your site's administrator to do so"

Happy to help out here to provide the specific links depending on context.

---

\* (we could even add a button to install it from right within that screen but that's a bit overkill)

**Comment by BrittanyIRL** [comment]
@swissspidy I might need a little help unveiling these nuances in php when you have time. Right now I've just got it to tell me if it's installed or active. I'm not sure how to tell if they have the GA module active or not.

Another question is if `current_user_can( 'manage_options' )` is what decides if a user has plugin installation permissions?

**Comment by swissspidy** [comment]
`current_user_can( 'install_plugins' )` :-)

No need to dig into the PHP stuff if it takes too long, just so you know!

**Comment by swissspidy** [comment]
> I'm not sure how to tell if they have the GA module active or not.

We actually do that already in `Analytics.php`. But since we now also need that in `Dashboard.php` Jonny & I probably have to figure out how to best share that code.

**Comment by BrittanyIRL** [comment]
Thanks! 

So, thoughts on making this a follow up ticket and assigning it to you/jonny to avoid affecting scope for you? 

```
If Site Kit is active, but GA module is not active

a) If the user has permissions to configure Site Kit, provide direct link to Sit Kit settings screen.
b) If user lacks these permissions (quite unlikely but possible), either provide link to Site Kit dashboard, or provide no link and write something like "Reach out to your site's administrator to do so"
```

and for the time being if site kit is active just don't link to anything and just display text to the effect of ` Site Kit by Google has already enabled Google Analytics for your Web Stories, all changes to your analytics tracking should occur there.` 

I can handle the other updates outside of how to tell if GA module is active or not.

**Comment by swissspidy** [comment]
Sounds good 👍

**Comment by BrittanyIRL** [comment]
i'll make the updates and then the follow up ticket once i can confirm the rest of it is doable :)

**Comment by BrittanyIRL** [comment]
@swissspidy  - 

Alrighty, here's a link to the follow up ticket to separate work/scope: https://github.com/google/web-stories-wp/issues/5051 

And my final question for this I believe is how you would suggest linking to the plugin within the admin. I am stumped because when you search for the plugin in the admin plugin search section and click more details the best I can get is a modal that doesn't impact the URL at all! I feel like I'm missing something obvious or misinterpreting what you mean.

**Comment by csossi** [comment]
Any QA Instructions that you can add here, @BrittanyIRL ?

**Comment by BrittanyIRL** [comment]
Sure, sorry about that - I thought you were referencing the "how to test" section of the PR. Here's testing/changes. Or see this PR description: https://github.com/google/web-stories-wp/pull/5049

### User-facing changes

Shows user 1 of three messages depending on if site kit is installed and conditionally disables tracking id input.
- Install Site Kit by Google to easily enable Google Analytics for Web Stories. (input enabled)
- Use Site Kit by Google to easily activate Google Analytics for Web Stories. (input disabled)
- Site Kit by Google has already enabled Google Analytics for your Web Stories, all changes to your analytics tracking should occur there. (input disabled)


### Testing Instructions
In your WordPress dashboard: install site kit and enable it and see messages change in your editor settings view (dashboard -> editor settings)

Installed and active:
https://user-images.githubusercontent.com/10720454/97620953-7833e380-19df-11eb-9d69-e4b5e4ab1e90.png

Installed and inactive
https://user-images.githubusercontent.com/10720454/97620744-34d97500-19df-11eb-94b5-1815a177670f.png

Not installed
https://user-images.githubusercontent.com/10720454/97620752-35720b80-19df-11eb-9e57-5f868974461a.png

**Comment by csossi** [comment]
These links are refreshing page - should open in new browser tabs:
![image.png](https://images.zenhubusercontent.com/5ea085e4704e72c0a99c6997/4828db03-6e2c-47cd-984b-5f13dfcb01d0)

This is true for all three states (except for "not installed" where "install site kit" refreshes page to plugin page)

**Comment by samitron7** [comment]
@BrittanyIRL I don't have SiteKit installed and not seeing the copies under the Google Analytics Tracking ID title:

![Screen Shot 2020-11-16 at 2.28.52 PM.png](https://images.zenhubusercontent.com/5e9dc6e34754fda9ccd26e86/db6ade15-d06d-4f2d-95a5-1b477df77297)
![Screen Shot 2020-11-16 at 2.29.06 PM.png](https://images.zenhubusercontent.com/5e9dc6e34754fda9ccd26e86/6a59cf23-8e15-4ab2-bb18-a3673e4cb383)

**Comment by swissspidy** [comment]
@samitron7 Note that the change hasn't been merged yet, so you won't see this in 1.1.1 of the plugin. Still in QA / progress stage.

**Comment by samitron7** [comment]
@swissspidy Ah apologizes, I thought I was viewing in UAT.

**Comment by BrittanyIRL** [comment]
> These links are refreshing page - should open in new browser tabs:
> ![image.png](https://camo.githubusercontent.com/c3af5f9e97657008e8a598162cb58241d1cdb0e10c10bddc2e8bade82b7b09cc/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3565613038356534373034653732633061393963363939372f34383238646230332d366532632d343763642d393834622d356631336466636230316430)
> 
> This is true for all three states (except for "not installed" where "install site kit" refreshes page to plugin page)

They are working fine for me when i test it in chrome and firefox. can you test again and if it's still occurring give me details about your set up so i can reproduce it? We're using a feature from the wordPress internationalization package to create these links within text, I'm thinking i needed to just rebase and get current for some update to possibly take that got out of sync in the time this PR has been open.

**Comment by swissspidy** [comment]
@BrittanyIRL are they all opening in a new tab/window

**Comment by BrittanyIRL** [comment]
> @BrittanyIRL are they all opening in a new tab/window

@swissspidy - nope they are redirecting from the existing tab

**Comment by swissspidy** [comment]
> > @BrittanyIRL are they all opening in a new tab/window
> 
> @swissspidy - nope they are redirecting from the existing tab

right, so I think Claudio suggested that these should open in a new tab if the link is pointing to another site (like WordPress.org). WDYT about that?
We could pass this information (whether link is external or not) from PHP to the app, so it could add `target=_blank` if needed

**Comment by BrittanyIRL** [comment]
ah that makes sense, i was misunderstanding. yup that works, i'll update. Thanks for mediating!

**Comment by BrittanyIRL** [comment]
ok, @csossi let's try this one more time. The links should all open a new tab now.

**Comment by csossi** [comment]
Verified in QA

**Comment by samitron7** [comment]
Verified in UAT

**Comment by felixarntz** [comment]
I recently ran into this while using the latest release, but then found this issue was already there and resolved 🎉 

Just curious about the following idea: It could be more intuitive to also display the Analytics tracking ID used by Site Kit in the field, maybe as `readonly`? I think that would be even more helpful contextually and more obvious at first sight. Right now the field is still available as usually as far as I can tell, which could lead to confusing expectations where users search for their ID and add it to the field although it's already taken care of by Site Kit.

Let me know if this is worth a follow-up issue.

cc @swissspidy

**Comment by swissspidy** [comment]
> Right now the field is still available as usually as far as I can tell, which could lead to confusing expectations where users search for their ID and add it to the field although it's already taken care of by Site Kit.

If Site Kit is installed, the field is disabled/readonly and can't be altered, with a message displayed saying "Use Site Kit by Google to easily activate Google Analytics for Web Stories". If a tracking ID was added before Site Kit was installed, that tracking ID will be displayed in the disabled field though.

Good idea to display the one from Site Kit though. Sounds like a reasonable enhancement 👍
